### DISCUSSION TITLE
New `_exist` and `_nexist` operators in App Filters

### DISCUSSION THREAD

**Discussion by t7tran** [directus/directus#12991] [Feature Requests]
What do you guys think about adding support for those? In our use cases, we would like to check if a form is new based on its `id` field. If the field `id` (or any dependent field) isn't present in the payload, none of the current operators apply.

**Comment by rijkvanzanten** [comment]
Would this be used in conditional fields / validation primarily? You wouldn't be able to filter on it when fetching data for example (though I guess at that point it'd be the same as `_null`) 🤔 

Also, there's already a `_submitted` which I believe does a very similar thing

**Reply by t7tran** [reply]
Correct, those operators wouldn't work on query filters. `_submitted` is semantically the same as `_exist`, whereas `_nexist` is trickier. Below is my current attempt that is working. `_exist` is simply a filler to make a complete pair.

```
# /directus/packages/shared/src/utils/generate-joi.ts
...
		if (operator === '_exist') {
			schema[key] = getAnySchema().exist();
		}

		if (operator === '_nexist') {
			options.requireAll = false; // XXX special case, needs to skip required() toggle later
			schema[key] = getAnySchema().forbidden();
		}
...
```

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
